src/sentry/tasks/files.py

```diff
@@ -88,3 +93,51 @@ def delete_unreferenced_blobs(blob_model, blob_index_model, blob_ids):
                 # Do nothing if the blob was deleted in another task, or
                 # if had another reference added concurrently.
                 pass
+
+
+@instrumented_task(
+    name="sentry.tasks.files.copy_to_control",
+    queue="files.copy",
+    default_retry_delay=60 * 5,
+    max_retries=MAX_RETRIES,
+    autoretry_for=(DatabaseError, IntegrityError),
+    acks_late=True,
+)
+def copy_file_to_control_and_update_model(
+    app_name: str,
+    model_name: str,
+    model_id: int,
+    file_id: int,
+    **kwargs,
+):
+    from sentry.models.files import ControlFile, ControlFileBlob, File
+
+    if ControlFileBlob._storage_config() is None:
+        return
+
+    if random.uniform(0, 1) > settings.SENTRY_FILE_COPY_ROLLOUT_RATE:
+        return
```

Comment:
It would be more efficient to do these checks *before* spawning tasks as with the current sampling rate 99% of the tasks will just burn CPU/network being spawned/started only to do no work. We can see what the volume is like in production and if we start backlogging we can shift this logic around.